httpcore5/src/main/java/org/apache/hc/core5/util/ReflectionUtils.java

```diff
@@ -72,4 +75,35 @@ public static int determineJRELevel() {
         return 7;
     }
 
+    /**
+     * @since 5.3
+     */
+    public static <T> SocketOption<T> getExtendedSocketOptionOrNull(final String fieldName) {
+        try {
+            final Class<?> extendedSocketOptionsClass = Class.forName("jdk.net.ExtendedSocketOptions");
+            final Field field = extendedSocketOptionsClass.getField(fieldName);
+            return (SocketOption)field.get((Object)null);
+        } catch (final Exception ignore) {
+            return null;
+        }
+    }
+
+    /**
+     * object can be ServerSocket or Socket
+     *
+     * @since 5.3
+     */
+    public static <T> void setOption(final T object, final String fieldName, final T value) {
```

Comment:
@ok2c whether  I should use `ReflectionUtils.callSetter` directly? 

But the method doesn't throw any exception, which is not suitable for me, I want an exception to be thrown if there is a problem.

I don't know whether I should change the logic of `ReflectionUtils.callSetter`: the method doesn't swallow the exception, which will have an impact on existing usage.